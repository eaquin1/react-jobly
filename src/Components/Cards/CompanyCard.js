import React from "react";
import {Link} from "react-router-dom"
import {
    Card, CardText, CardBody,
    CardTitle, Container
  } from 'reactstrap';
import defaultLogo from "../../Static/default-logo.png"
import "./Cards.scss"

function CompanyCard({item}) {
return (
   <Container className="Card-container">
       
        <Card className="Card" >
            
        <CardBody>
        <img src={ defaultLogo || item.logo_url} alt={`${item.name} Logo`} />
          <CardTitle className="font-weight-bold">
              <Link to={`/companies/${item.handle}`} key={item.handle}>{item.name}</Link></CardTitle>
          <CardText>{item.description}</CardText>
        </CardBody>
      </Card>
   </Container>
)
}

export default CompanyCard