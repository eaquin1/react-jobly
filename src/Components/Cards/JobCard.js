import React from "react";
import {
    Card,
    CardText,
    CardBody,
    CardTitle,
    CardSubtitle,
    Button,
} from "reactstrap";

function JobCard({ item }) {
    return (
        <Card>
            <CardBody>
                <CardTitle className="font-weight-bold">{item.title}</CardTitle>
                <CardSubtitle>Salary: {item.salary}</CardSubtitle>
                <CardText>Equity: {item.equity}</CardText>
                <Button>Apply</Button>
            </CardBody>
        </Card>
    );
}

export default JobCard;
