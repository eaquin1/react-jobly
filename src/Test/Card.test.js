import React from 'react';
import { render } from '@testing-library/react';
import CompanyCard from '../Components/Cards/CompanyCard';
import JobCard from '../Components/Cards/JobCard';
import { MemoryRouter } from 'react-router';

// Testing Company Card
describe('Testing the Company Card', () => {
  let item = {
    handle: 'rithm',
    name: 'Rithm School',
    description: 'Become an exceptional developer in 16 weeks.',
    logo_url:
      'https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg',
    num_employees: 32,
  };
  it('it renders CompanyCard without crashing', () => {
    render(
      <MemoryRouter>
        <CompanyCard item={item} />
      </MemoryRouter>
    );
  });
  it('matches snapshot for company', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard item={item} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

//Testing Job Card

describe('Testing the JobCard', () => {
  let item = { title: 'Job', salary: 1000000, equity: 0.1 };
  it('it renders JobCard without crashing', () => {
    render(
      <MemoryRouter>
        <JobCard item={item} />
      </MemoryRouter>
    );
  });

  it('matches snapshot for job', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <JobCard item={item} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});