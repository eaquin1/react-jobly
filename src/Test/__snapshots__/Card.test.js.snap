// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Company Card matches snapshot for company 1`] = `
<DocumentFragment>
  <div
    class="Card-container container"
  >
    <div
      class="Card card"
    >
      <div
        class="card-body"
      >
        <img
          alt="Rithm School Logo"
          src="default-logo.png"
        />
        <div
          class="font-weight-bold card-title"
        >
          <a
            href="/companies/rithm"
          >
            Rithm School
          </a>
        </div>
        <p
          class="card-text"
        >
          Become an exceptional developer in 16 weeks.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the JobCard matches snapshot for job 1`] = `
<DocumentFragment>
  <div
    class="Card-container container"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="font-weight-bold card-title"
        >
          Job
        </div>
        <div
          class="card-subtitle"
        >
          Salary: 1000000
        </div>
        <p
          class="card-text"
        >
          Equity: 0.1
        </p>
        <button
          class="float-right btn btn-secondary"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
